examples/beamforming/mvdr.py

```diff
@@ -0,0 +1,418 @@
+        eps (float): a value added to the denominator in mask normalization. Default: 1e-15
+    """
+
+    def __init__(self, normalize: bool = True, eps: float = 1e-15):
+        super().__init__()
+        self.normalize = normalize
+        self.eps = eps
+
+    def forward(self, X: torch.Tensor, mask: Optional[torch.Tensor] = None):
+        """
+        Args:
+            X (torch.Tensor): multi-channel complex-valued STFT matrix.
+                Tensor of dimension (..., freq, channel, time)
+            mask (torch.Tensor, optional): Time-Frequency mask for normalization.
+                Tensor of dimension (..., freq, time)
+
+        Returns:
+            psd torch.Tensor: PSD matrix of the input STFT matrix.
+                Tensor of dimension (..., freq, channel, channel)
+        """
+        # outer product: (..., channel_1, time) x (..., channel_2, time) -> (..., time, channel_1, channel_2)
+        psd_X = torch.einsum("...ct,...et->...tce", [X, X.conj()])
+
+        if mask is not None:
+            # TODO: future support for multi-channel mask
+
+            # Normalized mask along time dimension:
+            if self.normalize:
+                mask = mask / (mask.sum(dim=-1, keepdim=True) + self.eps)
+
+            psd = psd_X * mask.unsqueeze(-1).unsqueeze(-1)
+        else:
+            psd = psd_X
+
+        psd = psd.sum(dim=-3)
+        return psd
+
+
+class MVDR(torch.nn.Module):
+    """Basic MVDR module.
+
+    Args:
+        ref_channel (int, optional): the reference channel for beamforming. (Default: ``0``)
+        solution (str, optional): the solution to get MVDR weight.
+            Options: [``ref_channel``, ``stv_evd``, ``stv_power``]. (Default: ``ref_channel``)
+        diag_loading (bool, optional): whether apply diagonal loading on the psd matrix of noise
+            (Default: ``True``)
+        diag_eps (float, optional): the coefficient multipied to the identity matrix for diagonal loading
+            (Default: 1e-7)
+        online (bool, optional): whether to update the mvdr vector based on the previous psd matrices.
+            (Default: ``False``)
+    """
+
+    def __init__(
+        self,
+        ref_channel: int = 0,
+        solution: Optional[str] = "ref_channel",
+        diag_loading: bool = True,
+        diag_eps: float = 1e-7,
+        online: bool = False,
+    ):
+        super().__init__()
+        assert solution in ["ref_channel", "stv_evd", "stv_power"],\
+            "Unknown solution provided. Must be one of [``ref_channel``, ``stv_evd``, ``stv_power``]."
+        self.ref_channel = ref_channel
+        self.solution = solution
+        self.diag_loading = diag_loading
+        self.diag_eps = diag_eps
+        self.online = online
+        self.psd = torch.jit.trace(PSD(), (torch.rand(2, 129, 6, 100, dtype=torch.cdouble), torch.rand(2, 129, 100)))
+
+        psd_s: Optional[torch.Tensor] = torch.zeros(1)
```

Comment:
Is there a reason why these are of type `Optional`? I don't see a scenario where it equals to `None`, and it's typed as `torch.Tensor` in function inputs